src/scripts/generate_environment_profile.sh

```
@@ -0,0 +1,74 @@
+)
+
+# Default value for INGRESS_CONTROLLER_TYPE
+INGRESS_CONTROLLER_TYPE=""
+
+function compute_ingress_controller_type() {
+  local profile_name="$1"
+  local values_file="$2"
+
+  YQ_EXPR='select(.ingress.enabled == true) | .ingress.annotations."kubernetes.io/ingress.class" // ""'
+
+  local ingress_controller_class=$(yq "$YQ_EXPR" "$values_file")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)